src/worleynoise.rs

```diff
@@ -0,0 +1,81 @@
+use crate::error::Result;
+use rand::*;
+use std::fmt::Write;
+
+byond_fn! { worley_generate(density, threshold, width, height) {
+    worley_noise(density, threshold, width, height).ok()
+} }
+
+//In the worst possible situation this is O(n^3) in the best possible O(n^2.5), so just keep it in mind.
+fn worley_noise(    density_as_str: &str,
+    positive_threshold_as_str: &str,
+    width_as_str: &str,
+    height_as_str: &str,
+) -> Result<String> {
+    let density = density_as_str.parse::<f64>().unwrap(); // density of noise, 0 means no nodes and 100 means that every tile has a node.
```

Comment:
Can you replace all these unwraps with similar error handling to the other noise functions?

https://github.com/tgstation/rust-g/blob/c5cb44d3adc533951b8ea78638d65d74d88c0e45/src/noise_gen.rs#L19
https://github.com/tgstation/rust-g/blob/c5cb44d3adc533951b8ea78638d65d74d88c0e45/src/cellularnoise.rs#L17

It's still not a particularly good approach but in BYOND land nobody knows what to do when stuff crashes, so this works out better at least.

Reply:
yes sure